### ISSUE TITLE
[Feature] Deleting server.pid file on Docker startup

### ISSUE THREAD

**Issue by corbindavenport** [tootsuite/mastodon#6929]
About a week ago, I restored a backup of my server, which was made while Mastodon was running. When I tried to restart the server, I was met with this error:

```
web_1 | A server is already running. Check /mastodon/tmp/pids/server.pid.
```

As far as I understand the problem, as long as there is a file in that location, Ruby thinks the server is already running and won't start. After some trial and error, I discovered how to delete the file, by running this command:

```
docker exec -it mastodon_web_1 rm /mastodon/tmp/pids/server.pid
```

After that, the server started up normally. Would it be possible for Mastodon to run that command on startup (before the Ruby server is turned on), either by adding it to the `docker-compose.yml` file, or through some other method?

It appears [I'm not the only one that has had this problem](https://github.com/tootsuite/mastodon/issues/5940#issuecomment-350431568), and as someone who hasn't used Docker much, it wasn't immediately obvious how to fix that error (since it was in Docker's file system, and not the directory that the Git files are in). Hopefully this can save some people headaches down the road.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by Gargron** [comment]
Containers are replaceable (the data is in volumes) so you can do stuff like `docker-compose rm web` and then `docker-compose up -d` and it will re-create a fresh web container without the PID.

**Comment by corbindavenport** [comment]
Interesting, I'll have to make a note of that. But this is still a potential problem that should probably be fixed (possibly with a better method that I am not aware of).

**Comment by Gargron** [comment]
Fixed in #7052
